libbpf-cargo/src/btf/btf.rs

```diff
@@ -880,6 +880,18 @@ impl Btf {
                         agg_content.push(format!(r#"    pub {}: {},"#, field_name, field_ty_str));
                     }
 
+                    if t.is_struct && !packed {
```

Comment:
why `!packed` check? self.required_padding should take care of this?

not sure if it makes any difference, but it looks unnecessary and thus suspicious

Reply:
Yeah, it's not strictly necessary. I had it there to make the fact that no padding is desired when a struct is packed more explicit, as it's not obvious otherwise. But given that `required_padding` takes `packed` as an input that's probably good enough. Removed.

Reply:
>  I had it there to make the fact that no padding is desired when a struct is packed more explicit

You can have a packed struct where the last field ends way before the struct ends, so padding would still be necessary. E.g., try adding few more tests:

```
struct blah {
    char x;
    int y __attribute__((packed));
    long:64;
}
```

You can come up with some crazy cases. And if you manipulate BTF itself you can have even crazier ones.  Just look at https://patchwork.kernel.org/project/netdevbpf/patch/20221212211505.558851-7-andrii@kernel.org/ and the whole patch set.

Obviously, no one sane would be doing this explicitly in practice. But again, why making unnecessary simplifying assumptions, right?